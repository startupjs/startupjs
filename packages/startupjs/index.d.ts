export * from '@startupjs/react-sharedb'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline
export function css (): void
export function styl (): void
