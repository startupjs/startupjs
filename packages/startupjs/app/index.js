module.exports = require('@startupjs/app')
