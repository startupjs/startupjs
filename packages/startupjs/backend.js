export { default } from '@startupjs/backend'
export * from '@startupjs/backend'
