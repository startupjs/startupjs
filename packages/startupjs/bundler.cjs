throw new Error(`
[startupjs/bundler] ERROR! StartupJS bundler api has changed.

Now instead of using general \`require('startupjs/bundler.cjs')\` you have to require
a specific file you need.

Here is what you need to change:

1. \`require('startupjs/bundler.cjs').webpackWebConfig\`
    v v v
   \`require('startupjs/bundler/webpack.web.config.cjs')\`

2. \`require('startupjs/bundler.cjs').webpackServerConfig\`
    v v v
   \`require('startupjs/bundler/webpack.server.config.cjs')\`

3. \`require('startupjs/bundler.cjs').babelConfig\`
    v v v
   \`require('startupjs/bundler/babel.config.cjs')\`

4. \`require('startupjs/bundler.cjs').metroConfig\`
    v v v
   \`require('startupjs/bundler/metro.config.cjs')\`

5. \`require('startupjs/bundler.cjs').rnConfig\`
    v v v
   \`require('startupjs/bundler/react-native.config.cjs')\`

6. \`require('startupjs/bundler.cjs').viteConfig\`
    v v v
   \`require('startupjs/bundler/vite.config.cjs')\`

Additionally, if you did override .js/.jsx compilation in your webpack.web.config.cjs, you
might have used \`getJsxRule\` function there. In which case you should now get it from helpers:

7. \`const { getJsxRule } = require('startupjs/bundler.cjs')\`
    v v v
   \`const { getJsxRule } = require('startupjs/bundler/helpers.cjs')\`
`)
