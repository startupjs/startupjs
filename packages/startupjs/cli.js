#!/usr/bin/env node
'use strict'

import cli from '@startupjs/cli'
import { fileURLToPath } from 'url'
import path, { dirname } from 'path'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const templatesPath = path.join(__dirname, 'templates')

cli.run({ templatesPath })
