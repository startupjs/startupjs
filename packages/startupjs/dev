#!/bin/sh
# shellcheck disable=SC2059

# Exit on first failed command
set -e

# Run startupjs project using only Docker
#
# Version: 0.1.0
#
# Suports Mac, Linux, Windows WSL2
#
# Usage:
#   ./dev - universal command to run the main terminal or spawn additional terminals.
#           It tries to do the following:
#           1. Run the dev docker image of the same version of startupjs as in package.json.
#           2. If dev image is already running -- exec into it.

# Use current folder name as the name for container and the image by default
# ref: https://stackoverflow.com/a/43919044
a="/$0"; a=${a%/*}; a=${a#/}; a=${a:-.}; SCRIPT_DIR=$(cd "$a"; pwd)
FOLDER_NAME=$( basename "$SCRIPT_DIR" )

IMAGE_NAME="startupjs/dev"
CONTAINER_NAME="$FOLDER_NAME"

# Get startupjs version from package.json
STARTUPJS_VERSION=$( [ -f ./package.json ] && (grep -m 1 '"startupjs"' < ./package.json | sed -n 's/^[^"]*"[^"]*"[^"]*"//p' | sed -n 's/"[^"]*$//p') || echo '' )
STARTUPJS_VERSION=${STARTUPJS_VERSION:-latest}
# Remove leading ^ or ~ and remove the last patch number.
# This way ^0.40.2 will be converted to 0.40
# TODO: if yarn.lock exists, get the exact patch version from there
STARTUPJS_VERSION=$( echo "$STARTUPJS_VERSION" | sed -n 's/^[\^\~]*//p' | sed -n 's/\.*[0-9]*$//p' )

# Helper constants
ROOT="${PWD}"
# folder to store the MongoDB data
DB_FOLDER="${ROOT}/.mongo"
OS_NAME=$( uname -s | tr A-Z a-z )
# on Mac OS we use a faster driver for volume mounts
VOLUME_MODE=$( [ "$OS_NAME" = "darwin" ] && echo ":delegated" || echo "" )

# Colors
COLOR_NONE='\033[0m'
COLOR_LIGHT_GREEN='\033[1;32m'
COLOR_LIGHT_RED='\033[1;31m'

main () {
  if docker ps | grep "\s${CONTAINER_NAME}$"; then
    printf "${COLOR_LIGHT_GREEN}EXEC into the running container '${CONTAINER_NAME}'...${COLOR_NONE}\n"
    _exec
  else
    printf "${COLOR_LIGHT_GREEN}RUN a new dev container '${CONTAINER_NAME}'...${COLOR_NONE}\n"
    _cleanup
    _run
  fi
}

_exec () {
  printf "${COLOR_LIGHT_GREEN}You are now inside the dev environment. Run your commands here:${COLOR_NONE}\n"
  docker exec -it \
    -e "CONTAINER_NAME=${CONTAINER_NAME}" \
    "${CONTAINER_NAME}" \
    bash
}

_run () {
  PORT="${PORT:-$( _find_open_PORT )}"
  DEV_PORT="${DEV_PORT:-$( _find_open_DEV_PORT )}"
  if [ "$PORT" != "3000" ]; then
    printf "${COLOR_LIGHT_RED}IMPORTANT: Default port 3000 was occupied. Your app got assigned the port ${PORT}${COLOR_NONE}!\n"
  fi
  printf "${COLOR_LIGHT_GREEN}You are now inside the dev environment. Run your commands here:${COLOR_NONE}\n"
  echo "CONTAINER_NAME: $CONTAINER_NAME"
  echo "PORT: $PORT"
  echo "DEV_PORT: $DEV_PORT"
  echo "ROOT: $ROOT"
  echo "VOLUME_MODE: $VOLUME_MODE"
  echo "HOME: $HOME"
  echo "DB_FOLDER: $DB_FOLDER"
  echo "IMAGE_NAME: $IMAGE_NAME"
  echo "STARTUPJS_VERSION: $STARTUPJS_VERSION"
  docker run --rm -it \
    --name "${CONTAINER_NAME}" \
    --pull=always \
    -e "CONTAINER_NAME=${CONTAINER_NAME}" \
    -e "DOCKER_HOST_PORT=${PORT}" \
    --ulimit nofile=65535:65535 \
    -p "${PORT}:3000" \
    -p "${DEV_PORT}:3010" \
    -v "${ROOT}:/ws${VOLUME_MODE}" \
    -v "${HOME}/.ssh:/root/.ssh" \
    -v "${HOME}/.gitconfig:/root/.gitconfig" \
    -v "${DB_FOLDER}:/data/db${VOLUME_MODE}" \
    "${IMAGE_NAME}:${STARTUPJS_VERSION}"
}

# Keep only 5 latest startupjs images you pulled. Remove older ones
# ref: https://stackoverflow.com/a/40892314
_cleanup () {
  # shellcheck disable=SC2046
  docker rmi $(docker images -q ${IMAGE_NAME} | tail -n +6) || true
}

# a bit hardcody way to get available port for the app. Start with 3000 and go up to 3009
_find_open_PORT () {
  for iterator in 0 1 2 3 4 5 6 7 8 9; do
    a_port="300${iterator}"
    if ! lsof -n -i tcp:$a_port; then
      echo $a_port
      break
    fi
  done
}

# the same hardcody way as above to get available port for webpack devserver. Start with 3010 and go up to 3019
_find_open_DEV_PORT () {
  for iterator in 0 1 2 3 4 5 6 7 8 9; do
    a_port="301${iterator}"
    if ! lsof -n -i tcp:$a_port; then
      echo $a_port
      break
    fi
  done
}

main "$@"; exit
