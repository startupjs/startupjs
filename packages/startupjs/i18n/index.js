export * from '@startupjs/i18n'
