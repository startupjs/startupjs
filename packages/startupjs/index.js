export * from '@startupjs/react-sharedb'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline
export function css () {}
export function styl () {}
export { default as t } from '@startupjs/i18n/client/t.js'
