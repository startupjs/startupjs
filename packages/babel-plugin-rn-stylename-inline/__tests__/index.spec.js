const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {},
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  tests: {
    'Should throw an error if import is not specified': {
      error: true,
      code: `
        import React from 'react'

        styl\`
          .card
            color red
        \`
      `
    },
    'Should remove css and styl from startupjs import': /* js */`
      import React from 'react'
      import { css, observer, styl } from 'startupjs'
    `,
    'Should remove the whole lib if no other things are left in startupjs': /* js */`
      import React from 'react'
      import { css, styl } from 'startupjs'
    `,
    'Global css. Simple': /* js */`
      import React from 'react'
      import { css, observer } from 'startupjs'

      export default observer(function Card () {
        return <View styleName='card' />
      })

      css\`
        .card {
          color: red;
          background-color: green;
        }
      \`
    `,
    'Global styl. Simple': /* js */`
      import React from 'react'
      import { styl, observer } from 'startupjs'

      export default observer(function Card () {
        return <View styleName='card' />
      })

      styl\`
        .card
          color red
          background-color green
      \`
    `,
    'Global css. Several components': /* js */`
      import React from 'react'
      import { css } from 'startupjs'
      import { View } from 'react-native'

      export default function Card () {
        return <View styleName='card active'><Line /></View>
      }

      function Line () {
        return <View styleName='line' />
      }

      css\`
        .card {
          padding: 8px 16px;
        }
        .line {
          margin-top: 16px;
          border-radius: 8px;
        }
        .active {
          background-color: red;
        }
      \`
    `,
    'Global styl. Several components': /* js */`
      import React from 'react'
      import { styl } from 'startupjs'
      import { View } from 'react-native'

      export default function Card () {
        return <View styleName='card active'><Line /></View>
      }

      function Line () {
        return <View styleName='line' />
      }

      styl\`
        .card
          padding 8px 16px
        .line
          margin-top 16px
          border-radius 8px
        .active
          background-color red
      \`
    `,
    'Global and local css. Several components': /* js */`
      import React from 'react'
      import { css } from 'startupjs'
      import { View } from 'react-native'

      export default function Card () {
        return <View styleName='root active'><Line /></View>

        css\`
          .root {
            padding: 8px 16px;
          }
        \`
      }

      function Line () {
        return <View styleName='root' />

        css\`
          .root {
            margin-top: 16px;
            border-radius: 8px;
          }
        \`
      }

      css\`
        .active {
          background-color: red;
        }
      \`
    `,
    'Global and local css and styl. Several components': /* js */`
      import React from 'react'
      import { css, styl } from 'startupjs'
      import { View } from 'react-native'

      export default function Card () {
        return <View styleName='root active'><Line /></View>

        styl\`
          .root
            padding 8px 16px
        \`
      }

      function Line () {
        return <View styleName='root' />

        css\`
          .root {
            margin-top: 16px;
            border-radius: 8px;
          }
        \`
      }

      styl\`
        .active
          background-color red
      \`
    `,
    'Aliasing css and styl': /* js */`
      import React from 'react'
      import { css as myCss, styl as myStyl, observer } from 'startupjs'

      export default observer(function Card () {
        myStyl\`
          .card
            color blue
        \`
        return <View styleName='card' />
      })

      myCss\`
        .card {
          color: red;
          background-color: green;
        }
      \`
    `
  }
})
