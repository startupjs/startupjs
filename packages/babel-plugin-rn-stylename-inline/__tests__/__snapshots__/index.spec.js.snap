// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-rn-stylename-inline Aliasing css and styl: Aliasing css and styl 1`] = `

import React from 'react'
import { css as myCss, styl as myStyl, observer } from 'startupjs'

export default observer(function Card () {
  myStyl\`
    .card
      color blue
  \`
  return <View styleName='card' />
})

myCss\`
  .card {
    color: red;
    background-color: green;
  }
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { observer } from "startupjs";
const __CSS_GLOBAL__ = {
  card: {
    color: "red",
    backgroundColor: "green",
  },
  __hash__: 1762191192,
};
const _localCssInstance = {
  card: {
    color: "#00f",
  },
  __hash__: -1529996446,
};
export default observer(function Card() {
  const __CSS_LOCAL__ = _localCssInstance;
  return <View styleName="card" />;
});


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global and local css and styl. Several components: Global and local css and styl. Several components 1`] = `

import React from 'react'
import { css, styl } from 'startupjs'
import { View } from 'react-native'

export default function Card () {
  return <View styleName='root active'><Line /></View>

  styl\`
    .root
      padding 8px 16px
  \`
}

function Line () {
  return <View styleName='root' />

  css\`
    .root {
      margin-top: 16px;
      border-radius: 8px;
    }
  \`
}

styl\`
  .active
    background-color red
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { View } from "react-native";
const __CSS_GLOBAL__ = {
  active: {
    backgroundColor: "#f00",
  },
  __hash__: -1767660834,
};
const _localCssInstance2 = {
  root: {
    marginTop: 16,
    borderRadius: 8,
  },
  __hash__: -1053412432,
};
const _localCssInstance = {
  root: {
    paddingTop: 8,
    paddingRight: 16,
    paddingBottom: 8,
    paddingLeft: 16,
  },
  __hash__: -1823792365,
};
export default function Card() {
  const __CSS_LOCAL__ = _localCssInstance;
  return (
    <View styleName="root active">
      <Line />
    </View>
  );
}

function Line() {
  const __CSS_LOCAL__ = _localCssInstance2;
  return <View styleName="root" />;
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global and local css. Several components: Global and local css. Several components 1`] = `

import React from 'react'
import { css } from 'startupjs'
import { View } from 'react-native'

export default function Card () {
  return <View styleName='root active'><Line /></View>

  css\`
    .root {
      padding: 8px 16px;
    }
  \`
}

function Line () {
  return <View styleName='root' />

  css\`
    .root {
      margin-top: 16px;
      border-radius: 8px;
    }
  \`
}

css\`
  .active {
    background-color: red;
  }
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { View } from "react-native";
const __CSS_GLOBAL__ = {
  active: {
    backgroundColor: "red",
  },
  __hash__: -1812576046,
};
const _localCssInstance2 = {
  root: {
    marginTop: 16,
    borderRadius: 8,
  },
  __hash__: -1053412432,
};
const _localCssInstance = {
  root: {
    paddingTop: 8,
    paddingRight: 16,
    paddingBottom: 8,
    paddingLeft: 16,
  },
  __hash__: -1823792365,
};
export default function Card() {
  const __CSS_LOCAL__ = _localCssInstance;
  return (
    <View styleName="root active">
      <Line />
    </View>
  );
}

function Line() {
  const __CSS_LOCAL__ = _localCssInstance2;
  return <View styleName="root" />;
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global css. Several components: Global css. Several components 1`] = `

import React from 'react'
import { css } from 'startupjs'
import { View } from 'react-native'

export default function Card () {
  return <View styleName='card active'><Line /></View>
}

function Line () {
  return <View styleName='line' />
}

css\`
  .card {
    padding: 8px 16px;
  }
  .line {
    margin-top: 16px;
    border-radius: 8px;
  }
  .active {
    background-color: red;
  }
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { View } from "react-native";
const __CSS_GLOBAL__ = {
  card: {
    paddingTop: 8,
    paddingRight: 16,
    paddingBottom: 8,
    paddingLeft: 16,
  },
  line: {
    marginTop: 16,
    borderRadius: 8,
  },
  active: {
    backgroundColor: "red",
  },
  __hash__: 1310335761,
};
export default function Card() {
  return (
    <View styleName="card active">
      <Line />
    </View>
  );
}

function Line() {
  return <View styleName="line" />;
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global css. Simple: Global css. Simple 1`] = `

import React from 'react'
import { css, observer } from 'startupjs'

export default observer(function Card () {
  return <View styleName='card' />
})

css\`
  .card {
    color: red;
    background-color: green;
  }
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { observer } from "startupjs";
const __CSS_GLOBAL__ = {
  card: {
    color: "red",
    backgroundColor: "green",
  },
  __hash__: 1762191192,
};
export default observer(function Card() {
  return <View styleName="card" />;
});


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global styl. Several components: Global styl. Several components 1`] = `

import React from 'react'
import { styl } from 'startupjs'
import { View } from 'react-native'

export default function Card () {
  return <View styleName='card active'><Line /></View>
}

function Line () {
  return <View styleName='line' />
}

styl\`
  .card
    padding 8px 16px
  .line
    margin-top 16px
    border-radius 8px
  .active
    background-color red
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { View } from "react-native";
const __CSS_GLOBAL__ = {
  card: {
    paddingTop: 8,
    paddingRight: 16,
    paddingBottom: 8,
    paddingLeft: 16,
  },
  line: {
    marginTop: 16,
    borderRadius: 8,
  },
  active: {
    backgroundColor: "#f00",
  },
  __hash__: 553324671,
};
export default function Card() {
  return (
    <View styleName="card active">
      <Line />
    </View>
  );
}

function Line() {
  return <View styleName="line" />;
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Global styl. Simple: Global styl. Simple 1`] = `

import React from 'react'
import { styl, observer } from 'startupjs'

export default observer(function Card () {
  return <View styleName='card' />
})

styl\`
  .card
    color red
    background-color green
\`

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { observer } from "startupjs";
const __CSS_GLOBAL__ = {
  card: {
    color: "#f00",
    backgroundColor: "#008000",
  },
  __hash__: 772349652,
};
export default observer(function Card() {
  return <View styleName="card" />;
});


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Should remove css and styl from startupjs import: Should remove css and styl from startupjs import 1`] = `

import React from 'react'
import { css, observer, styl } from 'startupjs'

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";
import { observer } from "startupjs";


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Should remove the whole lib if no other things are left in startupjs: Should remove the whole lib if no other things are left in startupjs 1`] = `

import React from 'react'
import { css, styl } from 'startupjs'

      â†“ â†“ â†“ â†“ â†“ â†“

import React from "react";


`;

exports[`@startupjs/babel-plugin-rn-stylename-inline Should throw an error if import is not specified: Should throw an error if import is not specified 1`] = `

import React from 'react'

styl\`
  .card
    color red
\`

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: unknown: 
      Import not found for tagged template \`styl\`
    
[0m [90m 1 |[39m [36mimport[39m [33mReact[39m [36mfrom[39m [32m'react'[39m[0m
[0m [90m 2 |[39m[0m
[0m[31m[1m>[22m[39m[90m 3 |[39m styl[32m\`[39m[0m
[0m [90m   |[39m [31m[1m^[22m[39m[0m
[0m [90m 4 |[39m [32m  .card[39m[0m
[0m [90m 5 |[39m [32m    color red[39m[0m
[0m [90m 6 |[39m [32m\`[39m[0m

`;
